src/components/molecules/CheckboxGroupField/CheckboxGroupField.tsx

```diff
@@ -9,46 +9,53 @@ const CheckboxWrapper = styled.div`
   padding: 4px 0;
 `;
 
-interface ICheckboxGroupFieldItem {
+interface ICheckboxGroupFieldItem<Val extends Record<string, boolean>> {
```

Comment:
If you are using "Val" - you might as well use "Value" - abbreviations/nicknames aren't generally a good idea, good convo here, but also Uncle Bob called this out a while back in clean code https://dev.to/peledzohar/abbreviations-in-code-yay-or-nay-3001

Reply:
Or just use "V" I guess in the standard java/typescript generic nomenclature.

Reply:
The reason I did it is because I found it the same way in `formik` :)

Reply:
does it? interesting - in which case if its the convention they have chosen go with it. Not great but like someone said its better to be consistent